changelog/7206.bugfix.rst

```diff
@@ -0,0 +1 @@
+Fixed a bug with :func:`~sunpy.map.sample_at_coords()` where sampling outside the bounds of the map would sometimes not error and instead return strange pixel values.
```

Comment:
is there an ever a case where it would error?

```suggestion
Fixed a bug with :func:`~sunpy.map.sample_at_coords()` where sampling outside the bounds of the map would not error and instead return strange pixel values.
```

Reply:
Yes, the specific bug occurs when the array index is negative, but its absolute value is smaller than or equal to the size of that dimension.  Everywhere else out of bounds, there'd be an error.  To schematically illustrate:
```python
>>> import numpy as np
>>> x = np.arange(10)
>>> x[10]  # raises IndexError
>>> x[-11]  # raises IndexError
>>> x[-1]  # does not raise an error because it is interpreted as indexing from the end
```

Reply:
If you think of the original map as a patch on the sky, there are three patches on the sky of the same size that would not error.  The remaining 4 pi of the sky would error.

Reply:
ok, yes that makes sense - thanks